tests/providers/apache/beam/operators/test_beam.py

```diff
@@ -720,46 +749,148 @@ def test_init(self):
     @mock.patch("airflow.providers.apache.beam.operators.beam.GCSHook")
     def test_async_execute_should_execute_successfully(self, gcs_hook, beam_hook_mock):
         """
-        Asserts that a task is deferred and the BeamPipelineTrigger will be fired
+        Asserts that a task is deferred and the BeamPythonPipelineTrigger will be fired
         when the BeamRunPythonPipelineOperator is executed in deferrable mode when deferrable=True.
         """
         with pytest.raises(TaskDeferred) as exc:
             self.operator.execute(context=mock.MagicMock())
 
-        assert isinstance(exc.value.trigger, BeamPipelineTrigger), "Trigger is not a BeamPipelineTrigger"
+        assert isinstance(
+            exc.value.trigger, BeamPythonPipelineTrigger
+        ), "Trigger is not a BeamPythonPipelineTrigger"
 
-    def test_async_execute_should_throw_exception(self):
-        """Tests that an AirflowException is raised in case of error event"""
+    @mock.patch("airflow.providers.apache.beam.operators.beam.BeamHook")
+    @mock.patch("airflow.providers.apache.beam.operators.beam.GCSHook")
+    def test_async_execute_direct_runner(self, gcs_hook, beam_hook_mock):
+        """
+        Test BeamHook is created and the right args are passed to
+        start_python_workflow when executing direct runner.
+        """
+        gcs_provide_file = gcs_hook.return_value.provide_file
+        with pytest.raises(TaskDeferred):
+            self.operator.execute(context=mock.MagicMock())
+        beam_hook_mock.assert_called_once_with(runner=DEFAULT_RUNNER)
+        gcs_provide_file.assert_called_once_with(object_url=PY_FILE)
 
-        with pytest.raises(AirflowException):
-            self.operator.execute_complete(
-                context=mock.MagicMock(), event={"status": "error", "message": "test failure message"}
-            )
+    @mock.patch("airflow.providers.apache.beam.operators.beam.DataflowJobLink.persist")
```

Comment:
The only suggestion i have is to pass this "airflow.providers.apache.beam.operators.beam" as a separate variable and then pass it to each unit test :)
But it is perfect as it is already

Reply:
Good point! Thanks!